pontoon/base/signals.py

```diff
@@ -17,3 +22,78 @@ def project_locale_added(sender, **kwargs):
         project = project_locale.project
         project.has_changed = True
         project.save(update_fields=['has_changed'])
+
+
+@receiver(pre_save, sender=Locale)
+def create_locale_translators_group(sender, **kwargs):
+    """
+    Creates respective translators group in django's permission system.
+    Managers should be able to translate and manage locale.
+    """
+    instance = kwargs['instance']
+
+    if kwargs['raw'] or instance.translators_group:
+        return
+    try:
+        locale_ct = ContentType.objects.get(app_label='base', model='locale')
+        locale_group, _ = Group.objects.get_or_create(name='{} translators'.format(instance.code.upper()))
+        can_translate = Permission.objects.get(content_type=locale_ct, codename='can_translate_locale')
+        locale_group.permissions.add(can_translate)
+        instance.translators_group = locale_group
+    except:
```

Comment:
The difference is that a database entry not existing is a recoverable error that could happen and that we know how to fix, and is also something that the user can go on without. In that situation, it's appropriate for us to get a notification and fix it manually.

But the entire space of errors that could happen include errors that could corrupt data or break some part of the site if execution continues. That's why, in the general case, we want most exceptions to propagate up all the way to the error middleware, get sent to us, and show the user an error page.